client/my-sites/site-settings/seo-settings/form.jsx

```diff
@@ -62,91 +65,59 @@ function getGeneralTabUrl( slug ) {
 	return `/settings/general/${ slug }`;
 }
 
-function stateForSite( site ) {
-	return {
-		frontPageMetaDescription: get( site, 'options.advanced_seo_front_page_description', '' ),
-		isFetchingSettings: get( site, 'fetchingSettings', false ),
-	};
-}
-
-export class SeoForm extends Component {
-	static displayName = 'SiteSettingsFormSEO';
+export class SiteSettingsFormSEO extends Component {
+	_mounted = false;
 
 	state = {
-		...stateForSite( this.props.selectedSite ),
-		seoTitleFormats: this.props.storedTitleFormats,
-		// dirtyFields is used to prevent prop updates
-		// from overwriting local stateful edits that
-		// are in progress and haven't yet been saved
-		// to the server
 		dirtyFields: new Set(),
 	};
 
-	componentDidMount() {
-		this.refreshCustomTitles();
-	}
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		const { selectedSite: prevSite, isFetchingSite, translate } = this.props;
-		const { selectedSite: nextSite } = nextProps;
-		const { dirtyFields } = this.state;
+	static getDerivedStateFromProps( props, state ) {
+		const { dirtyFields } = state;
+		const nextState = {};
 
-		// save success
-		if ( this.state.isSubmittingForm && nextProps.isSaveSuccess ) {
-			this.props.markSaved();
-			this.props.requestSiteSettings( nextProps.siteId );
-			this.refreshCustomTitles();
-			this.setState( { isSubmittingForm: false } );
+		if (
+			! dirtyFields.has( 'seoTitleFormats' ) &&
+			! isEqual( props.storedTitleFormats, state.seoTitleFormats )
+		) {
+			nextState.seoTitleFormats = props.storedTitleFormats;
 		}
 
-		// save error
-		if ( this.state.isSubmittingForm && nextProps.saveError ) {
```

Comment:
What's happening with this error notice? Are we removing it intentionally?

Reply:
It used to be dispatched twice. One from the component and one from within the thunk. So I removed this one.

Reply:
Gotcha 👍